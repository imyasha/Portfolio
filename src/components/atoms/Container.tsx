import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  width: 1140px;
  max-width: 94%;
  margin: 0 auto;
`;

interface IProps {
  children: React.ReactNode;
}

const Container = ({ children }: IProps) => <Wrapper>{children}</Wrapper>;

Container.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Container;
