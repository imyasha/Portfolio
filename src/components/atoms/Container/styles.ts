import styled from 'styled-components';

const StyledContainer = styled.div`
	width: 94%;
	max-width: 55rem;
	margin: 0 auto;
`;

export { StyledContainer };
