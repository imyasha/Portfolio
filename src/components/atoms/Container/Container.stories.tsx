import { Container } from './Container';

export default {
	title: 'atoms/Container',
	component: Container,
};

export const ContainerStory = ({ children }: any) => (
	<Container>{children}</Container>
);

ContainerStory.storyName = 'Container';
ContainerStory.args = {
	children: <p>Page Content</p>,
};
