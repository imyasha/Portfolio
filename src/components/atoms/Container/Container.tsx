import { FC } from 'react';
import { StyledContainer } from './styles';

const Container: FC = ({ children }) => {
	return <StyledContainer>{children}</StyledContainer>;
};

export { Container };
