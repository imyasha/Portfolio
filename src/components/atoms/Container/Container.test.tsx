import { Container } from '../Container';
import { render } from '@testing-library/react';

describe('Container', () => {
	it('renders correctly', () => {
		const { container } = render(<Container>Page Content</Container>);
		expect(container).toMatchSnapshot();
	});
});
