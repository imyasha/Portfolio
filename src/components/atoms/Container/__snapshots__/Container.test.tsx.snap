// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
<div>
  <div
    class="styles__StyledContainer-mepagl-0 exxZos"
  >
    Page Content
  </div>
</div>
`;
